core/trino-spi/src/main/java/io/trino/spi/connector/RowChangeParadigm.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+public enum RowChangeParadigm
+{
+    /**
+     * A storage paradigm in which the connector can efficiently update individual columns.
+     */
+    CHANGE_ONLY_UPDATED_COLUMNS,
+
+    /**
+     * A paradigm that translates a changed row into a delete by rowId, and an insert of a
+     * new record, which will get a new rowId
```

Comment:
How are those new Ids created?

Reply:
The ids are created by the `FileWriter` when the insert file is written out.  I added text in the Javadoc making this clear.